### PR TITLE
Shared file edit

### PR DESCRIPTION
**Pull request by dmerrill6**
# Change log

- Fix bug when updating a shared file that it didn't reflect on the local device.

- Now updating a shared file will update the remote version. NOTE: This is currently not notifying the other members of the shared file so they still see the old version. Created a card for fixing this (https://app.clubhouse.io/terminalsystems/story/19164/notify-all-members-when-a-shared-file-gets-updated)

- Updated buckets client to pull files from remote client, local cache and DHT (DHT not currently implemented). Whichever finishes first is returned and cached for the next time that it is requested. If we want to start pinning these files we can do it on this cache (uses IPFS client).

- A few refactors.

[ch18845]

### CODE REVIEW

File: core/sync/sync.go
Comment (jsonsivar) [inline core/sync/sync.go:268]: Does `getOpenFileReverseKey` need to filter by a db ID as well? This was done on a previous PR so might have missed that but since we can have the same bucket name in different threads (and both my files and shared with me files will have `personal` as the name and for broader API usage this might be needed, but maybe could be post-alpha).

File: core/textile/secure_bucket_client.go
Comment (jsonsivar) [inline core/textile/secure_bucket_client.go:281]: This should be uncommented for now right?

**Review by jsonsivar — APPROVED** [review]
LGTM once the merge conflicts are fixed. Nice work. Also, as discussed on the call earlier we don't need to prioritize the card for notifying shared file edits since those will be remote right?

File: core/sync/sync.go
Comment (dmerrill6) [reply core/sync/sync.go]: Added `dbID` to the cache in a40be09

File: core/textile/secure_bucket_client.go
Comment (dmerrill6) [reply core/textile/secure_bucket_client.go]: Fixed in a40be09

**Comment by dmerrill6** [comment]
> LGTM once the merge conflicts are fixed. Nice work. Also, as discussed on the call earlier we don't need to prioritize the card for notifying shared file edits since those will be remote right?

For shared files we now have a cache to store it once it's opened, so after getting the notification about a shared file being updated, we should simply clear this cache and it will be pulled from the hub the next time.

**Pull request merged**
